### PR TITLE
Added tcp chat

### PR DESCRIPTION
**Pull request by nzhovniriv**

### CODE REVIEW

File: tcp_chat.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	go announcer()
+	for {
+		conn, err := listener.Accept()
+		if err != nil {
+			log.Print(err)
+			continue
```
Comment (8tomat8) [inline tcp_chat.go:91]: Dont forgot to close connections. You missed it in if block =)

File: tcp_chat.go [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+	"time"
+)
+
+type Message struct{ ip, message string }
+type Client chan<- Message
+
+var (
+	joining  = make(chan Client)  // for client, which joins the chat
+	leaving  = make(chan Client)  // for client, which leaves the chat
+	messages = make(chan Message) // for all messages
```
Comment (8tomat8) [inline tcp_chat.go:18]: You have only one static room?

Comment (nzhovniriv) [reply tcp_chat.go:18]: Yes I have one room. I was a little bit confused by it.

File: tcp_chat.go (hunk 1 above)
Comment (nzhovniriv) [reply tcp_chat.go:91]: I have missed it. Thanks.

File: tcp_chat.go [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+	leaving <- ch
+	messages <- Message{address, "--" + name + " has left the chat room at " + time.Now().Format("15:04:05") + "."}
+	conn.Close()
+}
+
+// write message to clients
+func writeMessage(conn net.Conn, ch <-chan Message, ip string) {
+	for msg := range ch {
+		if strings.Contains(msg.ip, "Initial message") || msg.ip != ip {
```
Comment (arbrix) [inline tcp_chat.go:74]: what if several users will have the same IP?

**Comment by arbrix** [comment]
Commands were not implemented

File: tcp_chat.go (hunk 3 above)
Comment (nzhovniriv) [reply tcp_chat.go:74]: I was thinking of it. I want to use UUID for each tcp connection instead of IP.

**Pull request merged**